.gitlab-ci.yml

```diff
@@ -284,13 +284,27 @@ doc:refman:
   dependencies:
     - build:base
 
+doc:refman:dune:
+  <<: *dune-template
+  stage: test
+  dependencies:
+    - build:egde:dune:dev
+  variables:
+    OPAM_SWITCH: edge
+    DUNE_TARGET: refman
+  artifacts:
+    name: "$CI_JOB_NAME"
+    paths:
+      - _build/default/doc/sphinx_build/html
+
 doc:ml-api:odoc:
+  <<: *dune-template
```

Comment:
This is not actually simplifying anything, don't use template when everything in it except for a trivial line gets overridden.

Reply:
Well, to me it servers a very important function here which is documentation; also it is a seed for future refactoring so I'd be in favor of keeping it as it doesn't create an obvious problem.

Reply:
What does it document?
As for future refactoring, don't anticipate it by making things more complicated.

Reply:
It documents this target is using a Dune-based build.

Reply:
>It documents this target is using a Dune-based build.

Not worth it IMO. Please change.

Reply:
Let's not block this PR on this unrelated change. @ejgallego you can push it again once the benefits are clearer so that @SkySkimmer doesn't oppose.

Reply:
I'm afraid we face a difference of opinion here. In particular I don't want to have N targets when we move from `make -f Makefile.dune` to `dune` itself; and doing a search for `dune-template` is important in my workflow.

Reply:
Then move everything except `make -f Makefile dune` from dune-template to build:dune since it's specialised for it.

Reply:
`dune-template` is already in `master`, thus apart from quite deep nitpicking indeed the comment is unrelated to this PR.

Reply:
Ok, let's see now.